lib/library.cpp

```diff
@@ -1889,6 +1910,11 @@ bool Library::isSmartPointer(const Token* tok) const
 
 const Library::SmartPointer* Library::detectSmartPointer(const Token* tok, bool withoutStd) const
 {
+    if (tok->tokType() != Token::Type::eType && tok->tokType() != Token::Type::eName) {
+        return nullptr;
+    }
+
+    // TODO: this is flawed and might result in lookups like "uint64_tflags_"
     std::string typestr = withoutStd ? "std::" : "";
     while (Token::Match(tok, "%name%|::")) {
```

Comment:
@chrchr-github could you please have a look at this? Thanks.

Reply:
Off the top of my head, you could try something like this
~~~c++
if (tok->str() == "::")
    tok = tok->next();
while (Token::Match(tok, "%name% ::")) {
    typestr += tok->str();
    typestr += "::";
    tok = tok->tokAt(2);
}
if (tok && tok->isName()) {
    typestr += tok->str();
}
~~~

Reply:
Thanks. That's probably as convoluted as what I would have come up with (if I had tried). I was hoping there might be something more straight forward I was missing.

Reply:
Thanks. That produces proper types to look up. Published in #7378.